ironmon_tracker/data/RandomizerLog.lua

```diff
@@ -205,6 +205,11 @@ function RandomizerLog.parseEvolutions(logLines)
 			evo = RandomizerLog.alternateNidorans(evo)
 			if RandomizerLog.PokemonNameToIdMap[evo] ~= nil then
 				table.insert(RandomizerLog.Data.Pokemon[pokemonId].Evolutions, RandomizerLog.PokemonNameToIdMap[evo])
+				-- Add pre-evolutions to the evolved Pokemon
+				if RandomizerLog.Data.Pokemon[RandomizerLog.PokemonNameToIdMap[evo]].PreEvolutions == nil then
+					RandomizerLog.Data.Pokemon[RandomizerLog.PokemonNameToIdMap[evo]].PreEvolutions = {}
+				end
+				table.insert(RandomizerLog.Data.Pokemon[RandomizerLog.PokemonNameToIdMap[evo]].PreEvolutions, pokemonId)
```

Comment:
Tracking pre-evos as they are read in from the file

Reply:
A small note on code legibility here. Since `RandomizerLog.PokemonNameToIdMap[evo]` is being used in more places and is quite the chunk to read, I'd say it's a good idea to assign it to a local variable prior to the first `if` condition here. In this way, the code can be understood a bit more quickly and clearly.

```lua
evo = RandomizerLog.alternateNidorans(evo)
local evoPokemonId = RandomizerLog.PokemonNameToIdMap[evo]
if evoPokemonId ~= nil then
	table.insert(RandomizerLog.Data.Pokemon[pokemonId].Evolutions, evoPokemonId)
	-- Add pre-evolutions to the evolved Pokemon
	if RandomizerLog.Data.Pokemon[evoPokemonId].PreEvolutions == nil then
		RandomizerLog.Data.Pokemon[evoPokemonId].PreEvolutions = {}
	end
	table.insert(RandomizerLog.Data.Pokemon[evoPokemonId].PreEvolutions, pokemonId)
```